### PR TITLE
AWS: Fix/Suppress ErrorProne warnings

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationMetadataOperations.java [hunk 1]
```diff
@@ -241,7 +241,7 @@ public void testAlterTableSetPropertiesNoDataPathAccess() {
       AssertHelpers.assertThrows("attempt to alter a table without ALTER permission should fail",
           ForbiddenException.class,
           "Glue cannot access the requested resources",
-          () -> updateProperties.commit());
```
Comment (kbendick) [inline aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationMetadataOperations.java:244]: Question (non-blocking): What warning does this give (if you remember)?

**Review by kbendick — APPROVED** [review]
LGTM.

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks for this!

File: aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationMetadataOperations.java (hunk 1 above)
Comment (nastra) [reply aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationMetadataOperations.java:244]:
```
/...iceberg/aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationMetadataOperations.java:244: warning: [LambdaMethodReference] Lambda should be a method reference
          () -> updateProperties.commit());
          ^
    (see https://github.com/palantir/gradle-baseline#baseline-error-prone-checks)
```

**Review by danielcweeks — APPROVED** [review]

**Pull request merged**